tiledb/sm/array/array.cc

```diff
@@ -796,10 +769,13 @@ const NDRange& Array::non_empty_domain() {
 }
 
 void Array::set_non_empty_domain(const NDRange& non_empty_domain) {
-  std::lock_guard<std::mutex> lock{mtx_};
   non_empty_domain_ = non_empty_domain;
 }
 
+MemoryTracker* Array::memory_tracker() {
```

Comment:
Now that this lives on the array, the readers can access this directly from the array and we don't need a pointer anymore.

Reply:
This is needed for the `FragmentMetadata` class that is mutating the memory tracker.